pkg/dataplane/controller.go

```diff
@@ -310,6 +329,98 @@ func (c *Controller) evaluatePacket(ctx context.Context, p *network.Packet) bool
 	return allowed
 }
 
+// syncNFTablesRules adds the necessary rules to process the first connection packets in userspace
+// and check if network policies must apply.
+// TODO: We can divert only the traffic affected by network policies using a set in nftables or an IPset.
+func (c *Controller) cleanConntrackEntries(ctx context.Context) error {
+	logger := klog.FromContext(ctx)
+
+	logger.Info("Cleaning conntrack entries")
+	start := time.Now()
+	defer func() {
+		logger.Info("Cleaning conntrack entries", "elapsed", time.Since(start))
+	}()
+
+	flows, err := vishnetlink.ConntrackTableList(vishnetlink.ConntrackTable, vishnetlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+
+	allPodIPs, divertAll, err := c.policyEngine.GetManagedIPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	ipset := sets.Set[string]{}
+	for _, ip := range allPodIPs {
+		ipset.Insert(ip.String())
+	}
+
+	filter := newConntrackFilter()
+
+	for _, flow := range flows {
+		// TODO: only evaluate the flows that are established
+		// TCP for now, as they are the only ones supported by the library.
+		// UDP is not supported yet, but it can be done by parsing the raw attributes
+		// new flows will be evaluated by the nfqueue logic.
```

Comment:
I'm not sure exactly what this is saying. It needs better punctuation and capitalization and grammar...

"TODO: only evaluate the flows that are established" sounds like you're saying "in the future, we should only evaluate the flows that are established", except then it seems like you're saying "currently we only evaluate established TCP flows but in the future we should evaluate established UDP flows too".

Also not clear whether "not supported ... it can be done" means "the library could be extended to do this" or "we could do this ourselves without support from the library if we needed to".

Also I understand what the last sentence means, but it doesn't seem like it belongs _here_.

Reply:
autogenerated stuff, agents complete sentences and I give up fighting at the second line